src/Build/Construction/ProjectRootElement.cs

```diff
@@ -712,6 +720,18 @@ internal ProjectExtensionsElement ProjectExtensions
         internal string LastDirtyReason
             => _dirtyReason == null ? null : String.Format(CultureInfo.InvariantCulture, _dirtyReason, _dirtyParameter);
 
+        /// <summary>
+        /// Initialize an in-memory empty ProjectRootElement instance that CANNOT be saved later.
+        /// The ProjectRootElement will not be marked dirty.
+        /// Uses the global project collection.
+        /// </summary>
+        internal static ProjectRootElement CreateNotDirty(ProjectRootElementCacheBase projectRootElementCache)
```

Comment:
I would prefer a more descriptive name like `CreateEphemeral`. Let's follow up with that once this is in `main` to not reset testing.

Reply:
i will address this in the follow-up PR